pkg/linkutils/link.go

```diff
@@ -0,0 +1,33 @@
+package linkutils
+
+import (
+	"crypto/md5"
+	"encoding/base64"
+	"encoding/hex"
+	"strconv"
+	"time"
+
+	"github.com/nimble-link/backend/models"
+)
+
+func IsDuplicateCustomAlias(customAlias string) bool {
+	link := models.FindLinkByCustomAlias(customAlias)
+	return link.ID != 0
+}
+
+func GenerateHashFrom(originalURL string, userID uint) string {
+	currentTime := time.Now().String()
+
+	md5Hashed := getMD5Hash(strconv.FormatUint(uint64(userID), 10) + currentTime + originalURL)
+	base64Hashed := getBase64Hash(md5Hashed)
+	return base64Hashed[0:6]
+}
+
+func getMD5Hash(text string) string {
+	hash := md5.Sum([]byte(text))
+	return hex.EncodeToString(hash[:])
```

Comment:
How about just 

```suggestion
	return hex.EncodeToString(hash)
```

Reply:
@namtx got this error when remove `[:]`: cannot use hash (type [16]byte) as type []byte in argument to hex.EncodeToString